models/instrument.py

```diff
@@ -838,6 +906,79 @@ def _get_header_values_converters(self):
         """
         return {}
 
+    @classmethod
+    def _get_fits_hdu_index_from_section_id(cls, section_id):
+        """
+        Translate the section_id into the index of the HDU in the FITS file.
+        For example, if we have an instrument with 10 CCDs, numbered 0 to 9,
+        the HDU list will probably contain a generic HDU at index 0,
+        and the individual section information in 1 through 10, so
+        the function should return section_id+1.
+        Another example could have section_id=A give an index 1,
+        and a section_id=B give an index 2 (so the function will read
+        from a dictionary to translate the values).
+
+        Parameters
+        ----------
+        section_id: int or str
+            The identifier of the section.
+
+        Returns
+        -------
+        index: int
+            The index of the HDU in the FITS file.
+            Note that the index is 0-based, as this value is
+            used in the astropy.io.fits functions/objects,
+            not in the native FITS format (which is 1-based).
+        """
+        cls.check_section_id(section_id)
+        return int(section_id) + 1
```

Comment:
This isn't going to work with string section_ids (which is probably what we need to use, since we can't count on the HDUs showing up in any particular order in the FITS file).

Reply:
...I see you overrode this in the DECam class.  Perhaps put a note in the documentation that an instrument needs to override this if it will ever get a section_id that isn't an integer, and if we can't assume that the mapping of section id to order in FITS file is 1:1?